tests/test_queue.py

```diff
@@ -1609,3 +1609,66 @@ async def workflow(x: int) -> int:
     # Change the global version, verify it works
     GlobalParams.app_version = future_version
     assert await handle.get_result() == input
+
+
+def test_queue_partitions(dbos: DBOS, client: DBOSClient) -> None:
+
+    blocking_event = threading.Event()
+    waiting_event = threading.Event()
+
+    @DBOS.workflow()
+    def blocked_workflow() -> str:
+        waiting_event.set()
+        blocking_event.wait()
+        assert DBOS.workflow_id
+        return DBOS.workflow_id
+
+    @DBOS.workflow()
+    def normal_workflow() -> str:
+        assert DBOS.workflow_id
+        return DBOS.workflow_id
+
+    queue = Queue("queue", partition_queue=True, worker_concurrency=1)
```

Comment:
Since we don't persist queue metadata like `partition_queue` info in the database, it's possible that one process sets this to `True` while the other one sets this to `False`... and that'll silently corrupt the queue. Can we find a way to enforce a check?

Reply:
There's no possibility of corruption--each process would respect the constraints defined on that process, just as if you set concurrency or rate limits different on different processes. Not recommended, and hard to do accidentally.